In [54]:
# import useful libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier,RandomForestRegressor
from sklearn.preprocessing import LabelEncoder
from sklearn.ensemble import AdaBoostClassifier,AdaBoostRegressor
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.model_selection import RandomizedSearchCV

In [55]:
# load dataset
df = pd.read_csv("data/online_shoppers_intention.csv")
df.head()

Administrative  Administrative_Duration  Informational  \
0               0                      0.0              0   
1               0                      0.0              0   
2               0                      0.0              0   
3               0                      0.0              0   
4               0                      0.0              0   

   Informational_Duration  ProductRelated  ProductRelated_Duration  \
0                     0.0               1                 0.000000   
1                     0.0               2                64.000000   
2                     0.0               1                 0.000000   
3                     0.0               2                 2.666667   
4                     0.0              10               627.500000   

   BounceRates  ExitRates  PageValues  SpecialDay Month  OperatingSystems  \
0         0.20       0.20         0.0         0.0   Feb                 1   
1         0.00       0.10         0.0         0.0   Feb                 2   
2         0.20       0.20         0.0         0.0   Feb                 4   
3         0.05       0.14         0.0         0.0   Feb                 3   
4         0.02       0.05         0.0         0.0   Feb                 3   

   Browser  Region  TrafficType        VisitorType  Weekend  Revenue  
0        1       1            1  Returning_Visitor    False    False  
1        2       1            2  Returning_Visitor    False    False  
2        1       9            3  Returning_Visitor    False    False  
3        2       2            4  Returning_Visitor    False    False  
4        3       1            4  Returning_Visitor     True    False

In [56]:
# gather datset information
display(df.shape)
print(100*"=")
display(df.info())
print(100*"=")
display(df.describe())
print(100*"=")
display(df.isnull().sum())

(12330, 18)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12330 entries, 0 to 12329
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Administrative           12330 non-null  int64  
 1   Administrative_Duration  12330 non-null  float64
 2   Informational            12330 non-null  int64  
 3   Informational_Duration   12330 non-null  float64
 4   ProductRelated           12330 non-null  int64  
 5   ProductRelated_Duration  12330 non-null  float64
 6   BounceRates              12330 non-null  float64
 7   ExitRates                12330 non-null  float64
 8   PageValues               12330 non-null  float64
 9   SpecialDay               12330 non-null  float64
 10  Month                    12330 non-null  object 
 11  OperatingSystems         12330 non-null  int64  
 12  Browser                  12330 non-null  int64  
 13  Region                   12330 non-null  int64  
 14  TrafficType           

None

Administrative  Administrative_Duration  Informational  \
count    12330.000000             12330.000000   12330.000000   
mean         2.315166                80.818611       0.503569   
std          3.321784               176.779107       1.270156   
min          0.000000                 0.000000       0.000000   
25%          0.000000                 0.000000       0.000000   
50%          1.000000                 7.500000       0.000000   
75%          4.000000                93.256250       0.000000   
max         27.000000              3398.750000      24.000000   

       Informational_Duration  ProductRelated  ProductRelated_Duration  \
count            12330.000000    12330.000000             12330.000000   
mean                34.472398       31.731468              1194.746220   
std                140.749294       44.475503              1913.669288   
min                  0.000000        0.000000                 0.000000   
25%                  0.000000        7.000000               184.137500   
50%                  0.000000       18.000000               598.936905   
75%                  0.000000       38.000000              1464.157214   
max               2549.375000      705.000000             63973.522230   

        BounceRates     ExitRates    PageValues    SpecialDay  \
count  12330.000000  12330.000000  12330.000000  12330.000000   
mean       0.022191      0.043073      5.889258      0.061427   
std        0.048488      0.048597     18.568437      0.198917   
min        0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.014286      0.000000      0.000000   
50%        0.003112      0.025156      0.000000      0.000000   
75%        0.016813      0.050000      0.000000      0.000000   
max        0.200000      0.200000    361.763742      1.000000   

       OperatingSystems       Browser        Region   TrafficType  
count      12330.000000  12330.000000  12330.000000  12330.000000  
mean           2.124006      2.357097      3.147364      4.069586  
std            0.911325      1.717277      2.401591      4.025169  
min            1.000000      1.000000      1.000000      1.000000  
25%            2.000000      2.000000      1.000000      2.000000  
50%            2.000000      2.000000      3.000000      2.000000  
75%            3.000000      2.000000      4.000000      4.000000  
max            8.000000     13.000000      9.000000     20.000000

Administrative             0
Administrative_Duration    0
Informational              0
Informational_Duration     0
ProductRelated             0
ProductRelated_Duration    0
BounceRates                0
ExitRates                  0
PageValues                 0
SpecialDay                 0
Month                      0
OperatingSystems           0
Browser                    0
Region                     0
TrafficType                0
VisitorType                0
Weekend                    0
Revenue                    0
dtype: int64

In [57]:
# checking datatypes
columns = df.columns.values 
for col in columns:
    if df[col].dtypes=="object" or df[col].dtypes=="bool":
        print(df[col].value_counts())
        print(100*"=")

May     3364
Nov     2998
Mar     1907
Dec     1727
Oct      549
Sep      448
Aug      433
Jul      432
June     288
Feb      184
Name: Month, dtype: int64
Returning_Visitor    10551
New_Visitor           1694
Other                   85
Name: VisitorType, dtype: int64
False    9462
True     2868
Name: Weekend, dtype: int64
False    10422
True      1908
Name: Revenue, dtype: int64


In [58]:
# preprocess the data
encoder = LabelEncoder()
df['Month'] = encoder.fit_transform(df['Month'])
df['Revenue'] = encoder.fit_transform(df['Revenue'])
df['VisitorType'] = encoder.fit_transform(df['VisitorType'])
df['Weekend'] = encoder.fit_transform(df['Weekend'])

In [59]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12330 entries, 0 to 12329
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Administrative           12330 non-null  int64  
 1   Administrative_Duration  12330 non-null  float64
 2   Informational            12330 non-null  int64  
 3   Informational_Duration   12330 non-null  float64
 4   ProductRelated           12330 non-null  int64  
 5   ProductRelated_Duration  12330 non-null  float64
 6   BounceRates              12330 non-null  float64
 7   ExitRates                12330 non-null  float64
 8   PageValues               12330 non-null  float64
 9   SpecialDay               12330 non-null  float64
 10  Month                    12330 non-null  int32  
 11  OperatingSystems         12330 non-null  int64  
 12  Browser                  12330 non-null  int64  
 13  Region                   12330 non-null  int64  
 14  TrafficType           

In [60]:
# input features and target features seperation
X = df.drop(columns=['Revenue','Informational_Duration'],axis=1)
y_revenue = df['Revenue']
y_duration = df['Informational_Duration']

Model Building for Revenue prediction

In [61]:
models = {
    'Random Forest': RandomForestClassifier(),
    'Ada boost': AdaBoostClassifier()
}

In [92]:
# spilit the data
X_train,X_test,y_train,y_test = train_test_split(X,y_revenue,test_size=0.2,random_state=42)
print(X_train.shape)
print(X_test.shape)
print(y_train.shape)

(9864, 16)
(2466, 16)
(9864,)


In [63]:
for i in range(len(list(models))):
    model=list(models.values())[i]
    model.fit(X_train,y_train)

    #Make Predictions
    y_pred=model.predict(X_test)
    # Calculate accuracy and confusion matrix
    accuracy = accuracy_score(y_test, y_pred)
    confusion = confusion_matrix(y_test, y_pred) 
    print(list(models.keys())[i])
    print(f"accuracy score : {accuracy}")
    print(f"confusion matrix : {confusion}") 

Random Forest
accuracy score : 0.8941605839416058
confusion matrix : [[1978   77]
 [ 184  227]]
Ada boost
accuracy score : 0.8832116788321168
confusion matrix : [[1955  100]
 [ 188  223]]


Using ensemble learning models Random Forest and Ada boost , Random Forest have more accuracy as compared to Ada boost
such as 89.21%

In [64]:
# hyper-tune the Random Forest to increase the accuracy score
## hypertuning the model using grid search cv
params={
    
    "n_estimators":[50,100,200],
    "criterion":["gini","entropy"],
    "max_depth":[3,5,10]
    
}

In [65]:
model = RandomForestClassifier()
cv=RandomizedSearchCV(model,param_distributions=params,scoring='accuracy',cv=5,verbose=3)
cv.fit(X_train,y_train)

Fitting 5 folds for each of 10 candidates, totalling 50 fits
[CV 1/5] END criterion=entropy, max_depth=5, n_estimators=200;, score=0.898 total time=   1.3s
[CV 2/5] END criterion=entropy, max_depth=5, n_estimators=200;, score=0.900 total time=   1.2s
[CV 3/5] END criterion=entropy, max_depth=5, n_estimators=200;, score=0.909 total time=   1.2s
[CV 4/5] END criterion=entropy, max_depth=5, n_estimators=200;, score=0.897 total time=   1.0s
[CV 5/5] END criterion=entropy, max_depth=5, n_estimators=200;, score=0.909 total time=   1.0s
[CV 1/5] END criterion=gini, max_depth=10, n_estimators=50;, score=0.903 total time=   0.3s
[CV 2/5] END criterion=gini, max_depth=10, n_estimators=50;, score=0.903 total time=   0.3s
[CV 3/5] END criterion=gini, max_depth=10, n_estimators=50;, score=0.915 total time=   0.3s
[CV 4/5] END criterion=gini, max_depth=10, n_estimators=50;, score=0.897 total time=   0.3s
[CV 5/5] END criterion=gini, max_depth=10, n_estimators=50;, score=0.905 total time=   0.3s
[CV 

RandomizedSearchCV(cv=5, estimator=RandomForestClassifier(),
                   param_distributions={'criterion': ['gini', 'entropy'],
                                        'max_depth': [3, 5, 10],
                                        'n_estimators': [50, 100, 200]},
                   scoring='accuracy', verbose=3)

In [66]:
cv.best_params_

{'n_estimators': 200, 'max_depth': 10, 'criterion': 'entropy'}

In [67]:
# model training with best suitable hyper-tuned parameters

bestmodel=RandomForestClassifier(n_estimators=50,max_depth=10,criterion='entropy',oob_score=True)
bestmodel.fit(X_train,y_train)
y_pred=bestmodel.predict(X_test)
best_accuracy=round(accuracy_score(y_test,y_pred)*100,2)
print(best_accuracy)

89.66


Best accuracy achieved is 89.62% for revenue_prediction

Model Building to find the Informational_Duration using the Ensemble learning algorithm


In [70]:
# spilit the data
from sklearn.preprocessing import StandardScaler
X_train,X_test,y_train,y_test = train_test_split(X,y_duration,test_size=0.2,random_state=42)
std_scalar = StandardScaler()

In [71]:
X_train = std_scalar.fit_transform(X_train)
X_test = std_scalar.transform(X_test)

In [72]:
models = {
    'Random Forest': RandomForestRegressor(),
    'Ada boost': AdaBoostRegressor()
}


import numpy as np
from sklearn.metrics import r2_score,mean_absolute_error,mean_squared_error
def evaluate_model(true, predicted):
    mae = mean_absolute_error(true, predicted)
    mse = mean_squared_error(true, predicted)
    rmse = np.sqrt(mean_squared_error(true, predicted))
    r2_square = r2_score(true, predicted)
    return mae, rmse, r2_square

trained_model_list=[]
model_list=[]
r2_list=[]


for i in range(len(list(models))):
    model=list(models.values())[i]
    model.fit(X_train,y_train)

    #Make Predictions
    y_pred=model.predict(X_test)
    # Calculate accuracy and confusion matrix
    mae, rmse, r2_square=evaluate_model(y_test,y_pred)

    print(list(models.keys())[i])
    model_list.append(list(models.keys())[i])

    print('Model Training Performance')
    print("RMSE:",rmse)
    print("MAE:",mae)
    print("R2 score",r2_square*100)

    r2_list.append(r2_square)
    
    print('='*35)
    print('\n')

Random Forest
Model Training Performance
RMSE: 117.59479415379396
MAE: 31.341671612886124
R2 score 36.65270514927856


Ada boost
Model Training Performance
RMSE: 156.60640876394814
MAE: 58.73543787845321
R2 score -12.34944787793295




In [77]:
# hyper-tune the Random Forest to increase the accuracy score
## hypertuning the model using grid search cv
params={
    
    "n_estimators":[50,100,200],
    "criterion":['squared_error', 'absolute_error', 'friedman_mse', 'poisson'],
    "max_depth":[3,5,10]
    
}



In [78]:
model = RandomForestRegressor()
cv=RandomizedSearchCV(model,param_distributions=params,scoring='accuracy',cv=5,verbose=3)
cv.fit(X_train,y_train)

Fitting 5 folds for each of 10 candidates, totalling 50 fits


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=friedman_mse, max_depth=10, n_estimators=50;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=friedman_mse, max_depth=10, n_estimators=50;, score=nan total time=   0.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=friedman_mse, max_depth=10, n_estimators=50;, score=nan total time=   0.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=friedman_mse, max_depth=10, n_estimators=50;, score=nan total time=   0.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=friedman_mse, max_depth=10, n_estimators=50;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=squared_error, max_depth=5, n_estimators=100;, score=nan total time=   0.8s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=squared_error, max_depth=5, n_estimators=100;, score=nan total time=   0.8s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=squared_error, max_depth=5, n_estimators=100;, score=nan total time=   0.8s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=squared_error, max_depth=5, n_estimators=100;, score=nan total time=   0.8s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=squared_error, max_depth=5, n_estimators=100;, score=nan total time=   0.7s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=poisson, max_depth=5, n_estimators=100;, score=nan total time=   0.7s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=poisson, max_depth=5, n_estimators=100;, score=nan total time=   0.8s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=poisson, max_depth=5, n_estimators=100;, score=nan total time=   0.7s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=poisson, max_depth=5, n_estimators=100;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=poisson, max_depth=5, n_estimators=100;, score=nan total time=   0.9s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=squared_error, max_depth=3, n_estimators=100;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=squared_error, max_depth=3, n_estimators=100;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=squared_error, max_depth=3, n_estimators=100;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=squared_error, max_depth=3, n_estimators=100;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=squared_error, max_depth=3, n_estimators=100;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=friedman_mse, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=friedman_mse, max_depth=3, n_estimators=200;, score=nan total time=   1.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=friedman_mse, max_depth=3, n_estimators=200;, score=nan total time=   1.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=friedman_mse, max_depth=3, n_estimators=200;, score=nan total time=   1.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=friedman_mse, max_depth=3, n_estimators=200;, score=nan total time=   1.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=poisson, max_depth=5, n_estimators=50;, score=nan total time=   0.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=poisson, max_depth=5, n_estimators=50;, score=nan total time=   0.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=poisson, max_depth=5, n_estimators=50;, score=nan total time=   0.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=poisson, max_depth=5, n_estimators=50;, score=nan total time=   0.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=poisson, max_depth=5, n_estimators=50;, score=nan total time=   0.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=poisson, max_depth=3, n_estimators=200;, score=nan total time=   1.2s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=poisson, max_depth=3, n_estimators=200;, score=nan total time=   1.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=poisson, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=poisson, max_depth=3, n_estimators=200;, score=nan total time=   1.1s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=poisson, max_depth=3, n_estimators=200;, score=nan total time=   1.4s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=poisson, max_depth=10, n_estimators=50;, score=nan total time=   0.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=poisson, max_depth=10, n_estimators=50;, score=nan total time=   0.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=poisson, max_depth=10, n_estimators=50;, score=nan total time=   0.5s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=poisson, max_depth=10, n_estimators=50;, score=nan total time=   0.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=poisson, max_depth=10, n_estimators=50;, score=nan total time=   0.6s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=squared_error, max_depth=10, n_estimators=100;, score=nan total time=   1.2s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=squared_error, max_depth=10, n_estimators=100;, score=nan total time=   1.2s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=squared_error, max_depth=10, n_estimators=100;, score=nan total time=   1.2s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=squared_error, max_depth=10, n_estimators=100;, score=nan total time=   1.1s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=squared_error, max_depth=10, n_estimators=100;, score=nan total time=   1.1s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 1/5] END criterion=squared_error, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 2/5] END criterion=squared_error, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 3/5] END criterion=squared_error, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 4/5] END criterion=squared_error, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py:778: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: 
Traceback (most recent call last):
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\model_selection\_validation.py", line 767, in _score
    scores = scorer(estimator, X_test, y_test)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 234, in __call__
    return self._score(
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_scorer.py", line 282, in _score
    return self._sign * self._score_func(y_true, y_pred, **self._kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\utils\_param_validation.py", line 192, in wrapper
    return func(*args, **kwargs)
  File "e:\ineuron_FSDS\BMI_prediction\venv\lib\site-packages\sklearn\metrics\_classification.py", line 

[CV 5/5] END criterion=squared_error, max_depth=3, n_estimators=200;, score=nan total time=   1.3s


RandomizedSearchCV(cv=5, estimator=RandomForestRegressor(),
                   param_distributions={'criterion': ['squared_error',
                                                      'absolute_error',
                                                      'friedman_mse',
                                                      'poisson'],
                                        'max_depth': [3, 5, 10],
                                        'n_estimators': [50, 100, 200]},
                   scoring='accuracy', verbose=3)

In [79]:
cv.best_params_

{'n_estimators': 50, 'max_depth': 10, 'criterion': 'friedman_mse'}

In [83]:
bestmodel=RandomForestRegressor(n_estimators=50,max_depth=10,criterion='friedman_mse')
bestmodel.fit(X_train,y_train)
y_pred=bestmodel.predict(X_test)


In [84]:
def evaluate_model(true, predicted):
    mae = mean_absolute_error(true, predicted)
    mse = mean_squared_error(true, predicted)
    rmse = np.sqrt(mean_squared_error(true, predicted))
    r2_square = r2_score(true, predicted)
    return mae, rmse, r2_square


mae, rmse, r2_square=evaluate_model(y_test,y_pred)

In [86]:
print(r2_square)

0.37712613059653144


The best accuracy achieved using Random Forest with hypertuned parameters is 37.7%